applications/FluidDynamicsApplication/symbolic_generation/compressible_navier_stokes/compressible_navier_stokes_explicit_cpp_template_with_integration.cpp

```diff
@@ -1535,6 +1535,22 @@ void CompressibleNavierStokesExplicit<3,4>::CalculateMassMatrix(
     rMassMatrix *= GetGeometry().Volume();
 }
 
+template <unsigned int TDim, unsigned int TNumNodes>
+void CompressibleNavierStokesExplicit<TDim, TNumNodes>::CalculateLumpedMassVector(
+    VectorType& rLumpedMassVector,
+    const ProcessInfo& rCurrentProcessInfo) const
+{
+    // Initialize the lumped mass vector
+    constexpr IndexType size = TNumNodes * BlockSize;
+    if (rLumpedMassVector.size() != BlockSize) {
+        rLumpedMassVector.resize(size, false);
+    }
+
+    // Fill the lumped mass vector
+    const double nodal_mass = GetGeometry().DomainSize() / TNumNodes;

```

Comment:
Why not multiplying by volume/area?

Reply:
`DomainSize` gives you back either the volume or area according to the geometry working space dimension. That is, area for 2D geometries and volume for 3D volumetric geometries.

Reply:
Thanks for specifying.